python/plugins/processing/ProcessingPlugin.py

```diff
@@ -282,6 +289,88 @@ def initGui(self):
         self.iface.actionToggleEditing().triggered.connect(partial(self.sync_in_place_button_state, None))
         self.sync_in_place_button_state()
 
+        self.projectProvider = QgsApplication.instance().processingRegistry().providerById("project")
+        self.projectProvider.algorithmsLoaded.connect(self.updateProjectModelMenu)
+        self.projectModelsMenu = None
+        self.projectMenuAction = None
+        self.projectMenuSeparator = None
+
+    def executeProjectModel(self, alg_id, as_batch=False):
+        """Executes a project model"""
+
+        alg = QgsApplication.instance().processingRegistry().createAlgorithmById(alg_id)
+
+        if alg is not None:
```

Comment:
This is almost 1:1 copy of the existing code -- could you move this out to a common place and avoid the duplication? Both the Models menu and the toolbox should be able to call a common function here without either of them relying on the other.